truetype/truetype.go

```diff
@@ -240,6 +260,48 @@ func (f *Font) parseCmap() error {
 			offset += 2
 		}
 		f.cmapIndexes = f.cmap[offset:]
+
+		// If unable to parse cmap correctly, use unitype cmap parse implementation.
+		if len(f.cmapIndexes) < 1 {
+			charcodeMap := make(map[uint8]uint32, f.nGlyph)
+
+			for i := 0; i < segCount; i++ {
+				c1 := f.cm[i].start
+				c2 := f.cm[i].end
+				d := f.cm[i].delta
+				rangeOffset := f.cm[i].offset
+
+				for c := c1; c <= c2; c++ {
+					var gid uint32
+
+					if rangeOffset == 0 {
+						gid = (c + d) & 0xFFFF
+					} else {
+						index := int(uint16(rangeOffset)/2 + (uint16(c) - uint16(c1)) + uint16(i) - uint16(offset))
+
+						if index >= len(f.charcodeToGID) {
+							return FormatError("out of bounds")
+						}
+						if f.charcodeToGID[uint8(index)] != 0 {
+							gid = (f.charcodeToGID[uint8(index)] + d) & 0xFFFF
+						} else {
+							gid = 0
+						}
+					}
+
+					if gid > 0 {
+						//b := runeDecoder.ToBytes(uint32(c))
+						//r := runeDecoder.DecodeRune(b)
+						if int(gid) >= int(f.nGlyph) {
+							return FormatError("out of bounds")
+						}
+						b := byte(c)
+						charcodeMap[b] = gid
+					}
+				}
+			}
+			f.charcodeToGID = charcodeMap
```

Comment:
this is only set if the previous implementation fails?  So its kind of a patch for the current implementation?

Reply:
Yes, its patch.